src/app/shared/components/template/components/qr-code/qr-code.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, OnInit } from "@angular/core";
+import { TemplateBaseComponent } from "../base";
+import QRCode from "qrcode";
+
+@Component({
+  selector: "plh-qr-code",
+  templateUrl: "./qr-code.component.html",
+  styleUrls: ["./qr-code.component.scss"],
+})
+export class TmplQRCodeComponent extends TemplateBaseComponent implements OnInit {
+  value: string;
+  QRCodeDataURL: string;
+
+  ngOnInit() {
+    this.value = this._row.value;
+    this.generateQR(this.value);
+  }
+
+  async generateQR(inputString: string) {
```

Comment:
nit(blocking)
You already have these methods in the pipe so no need to repeat here. That way all you need is the `<img` tag with the src piped from the qr pipe and `_row.value`, and can omit all the logic from this component file

The benefit of doing that way also means that the qrcode will update if the input value updates